### PR TITLE
Add is on main thread

### PR DESCRIPTION
**Pull request by david-laundav**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
This is a new feature which allow developers to check whether they currently execute a task on the main thead or not.

### :boom: Does this PR introduce a breaking change?
No, no breaking changes.

### :memo: Links to relevant issues/docs
https://github.com/MvvmCross/MvvmCross/issues/2791

### :thinking: Checklist before submitting

- [X] All projects build
- [ ] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Rebased onto v6.1 branch

### CODE REVIEW

**Comment by david-laundav** [comment]
@Cheesebaron FYI.

File: MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs [hunk 1]
```diff
@@ -7,8 +7,17 @@
 
 namespace MvvmCross.Base
 {
-    public abstract class MvxMainThreadAsyncDispatcher : MvxMainThreadDispatcher, IMvxMainThreadAsyncDispatcher
+    public class MvxMainThreadAsyncDispatcher : MvxMainThreadDispatcher, IMvxMainThreadAsyncDispatcher
     {
+        private int managedThreadId;
+
+        public MvxMainThreadAsyncDispatcher(int managedThreadId)
```
Comment (martijn00) [inline MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs]: Where does this ctor get set?

File: MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs [hunk 2]
```diff
@@ -7,8 +7,17 @@
 
 namespace MvvmCross.Base
 {
-    public abstract class MvxMainThreadAsyncDispatcher : MvxMainThreadDispatcher, IMvxMainThreadAsyncDispatcher
+    public class MvxMainThreadAsyncDispatcher : MvxMainThreadDispatcher, IMvxMainThreadAsyncDispatcher
     {
+        private int managedThreadId;
```
Comment (Cheesebaron) [inline MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs]: We use `_` prefix for private member variables.

File: MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs [hunk 3]
```diff
@@ -7,8 +7,17 @@
...
+
+        public MvxMainThreadAsyncDispatcher(int managedThreadId)
+        {
+            this.managedThreadId = managedThreadId;
```
Comment (Cheesebaron) [inline MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs]: Remove `this.`

File: MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs (hunk 1 above)
Comment (Cheesebaron) [reply MvvmCross/Base/MvxMainThreadAsyncDispatcher.cs]: Doesn't seem like this is set anywhere as @martijn00 points out.

**Comment by Cheesebaron** [comment]
Could you maybe also take a look at the things CodeFactor complains about?

**Comment by nickrandolph** [comment]
@david-laundav I sent you a PR suggesting some changes. My thoughts are:
- Need to revert change to API (ie return bool not void) - this should be picked up later when we clear out the synchronous dispatcher
- Need to use the existing tests to see which thread we're on (see my PR for this)
- Need to include sample code in the playground that shows IsOnMainThread works for each platform

**Comment by nickrandolph** [comment]
@david-laundav did you get a chance to review the PR I sent you?

**Comment by david-laundav** [comment]
@nickrandolph Sorry, busy days. I haven't received a PR from you. Perhaps you sent it to the wrong user?

**Pull request closed without merging**